Search command + autocomplete with view

bsrecker opened the pull request:
Implemented search command that returns tags in the database that might be similar. 

Added autocomplete to view + implemented function to efficiently grab and cache tags.

Minor formatting changes. #9

bsrecker committed:
Implemented search command that returns tags in the database that might be similar.

Added autocomplete to view + implemented function to efficiently grab and cache tags.

Minor formatting changes.

Infernum1 commented on leaf/extensions/tags.py:
```diff
@@ -23,6 +27,46 @@ async def check_permissions(
+    async def tag_autocomplete(
+        self, interaction: discord.Interaction, current: str
+    ) -> List[app_commands.Choice[str]]:
+        if not current:
```
is this needed? will current ever be `None` since you're not setting a default value for it? I'm bad with app_commands so not completely sure but yeah

bsrecker replied:
Agreed, this is not needed.

Infernum1 reviewed (approved):
apart from that, lgtm at a glance

bsrecker committed:
Implemented search command that returns tags in the database that might be similar.

Added autocomplete to view + implemented function to efficiently grab and cache tags.

Minor formatting changes.

bsrecker committed:
Implemented search command that returns tags in the database that might be similar.

Added autocomplete to view + implemented function to efficiently grab and cache tags.

Minor formatting changes.

mudkipdev commented:
It should probably prompt the user with a list of the top 5 most used tags in the guild if they don't have anything typed.

bsrecker commented:
> It should probably prompt the user with a list of the top 5 most used tags in the guild if they don't have anything typed.

That could be an additional issue. That would probably require a bit more work.

Infernum1 commented:
I think we can merge it for now
